cranelift/codegen/src/isa/aarch64/inst/args.rs

```diff
@@ -645,3 +648,31 @@ impl VectorSize {
         }
     }
 }
+
+//=============================================================================
+// Instruction sub-components: atomic memory update operations
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(u8)]
+pub enum LLSCOp {
+    Add,
+    Sub,
+    And,
+    Or,
+    Xor,
+    Xchg,
+}
+
+impl LLSCOp {
+    // This is a bit silly.  Maybe it would be easier just to store an `AtomicRmwOp`
+    // directly in Inst::LLSC ?
```

Comment:
On investigation, I am inclined to leave this as-is (although rename it to `AtomicRMWOp`) for two reasons:
* `ALUOp` has 32- and 64-bit variants of each operation.  But `Inst::AtomicRMW` also has a `type` field, and so that gives the possibility of creating nonsensical combinations (it violates the no-junk no-confusion rule).  Eg, `type` indicating 64-bit, but with `op` being `ALUOp::Orr32`.
* per @bjorn3's comment above ("cg_clif also requires nand, max, min, umax, umin."), `ALUOp` doesn't have max/min/umax/umin.